src/lib/core/fiber.h

```diff
@@ -338,6 +338,23 @@ fiber_join(struct fiber *f);
 
 /**
  * Wait until the fiber is dead or timeout exceeded.
+ * Return 0 on success, or -1 and set diag in case of error.
+ * If @a retval is not NULL, then copy fiber function return value
+ * into the location pointed to by @a retval.
+ * The fiber must not be detached (@sa fiber_set_joinable()).
+ * @pre FIBER_IS_JOINABLE flag is set.
+ *
+ * \param f fiber to be woken up
+ * \param timeout time during which we wait for the fiber completion
+ * \param retval pointer to the variable for the fiber's return value
+ * \return 0 on success, -1 on error
+ */
+int
+fiber_join_timeout_internal(struct fiber *f, double timeout, int *retval);
```

Comment:
You placed this function into the public API section (starts with `/** \cond public */`). If so, then it should not have `_internal` suffix, should have `API_EXPORT` attribute, be exported as a symbol, needs a docbot request, and shouldn't mention `FIBER_IS_JOINABLE` flag (fiber flags are not exposed to the public API anywhere). Otherwise I would recommend to move it out of the public API.

In either case the function needs unit tests in this commit.

Reply:
Removed this function.